### PR TITLE
rizin: 0.6.3 -> 0.7.2, cutter: 2.3.4 -> 2.3.4

### PR DESCRIPTION
**Pull request by Mic92**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 7c6f434c** [comment]
Hmm, Darwin-only multiple-output reference loop? How interesting…

**Comment by Mic92** [comment]
But's in jasper from the looks of it.

**Comment by chayleaf** [comment]
could you update the plugins as well (rz-ghidra, jsdec, and I guess sigdb)? at this point there should probably be a common update script to update rizin and all related packages

**Comment by 7c6f434c** [comment]
> But's in jasper from the looks of it.

Still, interesting, even if for another package. Agreed that not a problem here.

**Review by mjoerg — COMMENTED** [review]
[Rizin v0.7.2](https://github.com/rizinorg/rizin/releases/tag/v0.7.2) is available

File: pkgs/development/tools/analysis/rizin/default.nix [hunk 1]
```diff
@@ -26,11 +25,11 @@
...
-  version = "0.6.3";
+  version = "0.7.1";
 
   src = fetchurl {
     url = "https://github.com/rizinorg/rizin/releases/download/v${version}/rizin-src-v${version}.tar.xz";
-    hash = "sha256-lfZMarnm2qnp+lY0OY649s206/LoFNouTLlp0x9FCcI=";
+    hash = "sha256-FJ3I7tQHAIm25OZQcdVfVxwNLkxy0u5CBWKiMhMIwpQ=";
   };
 
   mesonFlags = [
```
Comment (chayleaf) [inline pkgs/development/tools/analysis/rizin/default.nix:35]:
```suggestion
  mesonFlags = [
    "-Duse_sys_libzstd=enabled"
    "-Duse_sys_pcre2=enabled"
```

(except please put `use_sys_libzstd` after `use_sys_lz4` and `use_sys_pcre2` after `use_sys_tree_sitter`, the way it's ordered in meson_options.txt, I can't do it with Github suggestions)

**Review by bendlas — APPROVED** [review]

**Comment by Mic92** [comment]
Also fixed macOS compilation.

**Comment by kennystrawnmusic** [comment]
Can confirm by working off my own fork and adding these changes to it that this fixes both #294438 and #293783. So, the more quickly it can be merged the better.

**Comment by chayleaf** [comment]
[cutter: 2.3.4 -> 2.3.4](https://github.com/NixOS/nixpkgs/pull/293264/commits/8d5d9861bd122476874619d1fc52d8269ccf3893) might not have been the best commit message...

**Comment by Mic92** [comment]
That's an issue with nix-update, when I am re-running it on a half-updated package with autocommit enabled.

**Pull request merged**